test/Microsoft.AspNetCore.Authentication.Test/Cookies/CookieMiddlewareTests.cs

```diff
@@ -851,68 +740,85 @@ private Task SignOutAsWrong(HttpContext context)
         }
 
         [Fact]
-        public async Task MapWillNotAffectChallenge()
+        public async Task MapWillAffectChallenge()
         {
             var builder = new WebHostBuilder()
-                .Configure(app =>
-                {
-                    app.UseCookieAuthentication(new CookieAuthenticationOptions
-                    {
-                        LoginPath = new PathString("/page")
-                    });
-                    app.Map("/login", signoutApp => signoutApp.Run(context => context.Authentication.ChallengeAsync("Cookies", new AuthenticationProperties() { RedirectUri = "/" })));
-                })
-                .ConfigureServices(services => services.AddAuthentication());
+                .Configure(app => app.Map("/login", signoutApp => signoutApp.Run(context => context.ChallengeAsync("Cookies", new AuthenticationProperties() { RedirectUri = "/" }))))
+                .ConfigureServices(s => s.AddCookieAuthentication(o => o.LoginPath = new PathString("/page")));
             var server = new TestServer(builder);
 
             var transaction = await server.SendAsync("http://example.com/login");
 
             Assert.Equal(HttpStatusCode.Redirect, transaction.Response.StatusCode);
 
             var location = transaction.Response.Headers.Location;
-            Assert.Equal("/page", location.LocalPath);
+            Assert.Equal("/login/page", location.LocalPath);
             Assert.Equal("?ReturnUrl=%2F", location.Query);
         }
 
-        [Fact]
+        [ConditionalFact(Skip = "Revisit, exception no longer thrown")]
```

Comment:
I think the old test threw because it was doing an AutomaticChallenge followed by an second challenge, which no longer happens in the new stack, I'll have to debug the old test, to see what was triggering it